### PR TITLE
Tech debt/setup tests in pipeline

### PR DESCRIPTION
**Pull request by Faizi52996**
### Context
[AB#201673](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/201673)

### Change proposed in this pull request
to replace the variables in appsettings with library group

### Checklist
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
~~- [ ] You have run all unit/integration tests and they pass~~
- [x] Your branch has been rebased onto main
~~- [ ] You have tested by running locally~~

### CODE REVIEW

File: web/pipelines/build.yaml [hunk 1]
```diff
@@ -110,6 +110,7 @@ stages:
     condition: and(succeeded(), eq(variables['ShouldDeploy'],'true'))
     variables:
       - group: 'dsi pre-prod'
+      - group: 'environment variables'
```
Comment (jrabbott) [inline web/pipelines/build.yaml:113]: Please give this a more meaningful name.

File: web/pipelines/steps-web-run-a11y-tests.yaml [hunk 2]
```diff
@@ -8,3 +8,12 @@ steps:
     inputs:
       archiveFilePatterns: '${{ parameters.workspaceDir }}/a11y-tests/a11y-tests.zip'
       destinationFolder: '${{ parameters.workspaceDir }}/a11y-tests-files'
+  - task: qetza.replacetokens.replacetokens-task.replacetokens@3
```
Comment (jrabbott) [inline web/pipelines/steps-web-run-a11y-tests.yaml:11]: Please use latest version of this task `qetza.replacetokens.replacetokens-task.replacetokens@6`.

Examples can be found in `steps-run-terraform.yaml`

File: web/pipelines/steps-web-run-e2e-tests.yaml [hunk 3]
```diff
@@ -8,3 +8,11 @@ steps:
     inputs:
       archiveFilePatterns: '${{ parameters.workspaceDir }}/e2e-tests/e2e-tests.zip'
       destinationFolder: '${{ parameters.workspaceDir }}/e2e-tests-files'
+  - task: qetza.replacetokens.replacetokens-task.replacetokens@3
```
Comment (jrabbott) [inline web/pipelines/steps-web-run-e2e-tests.yaml:11]: Please use latest version of this task `qetza.replacetokens.replacetokens-task.replacetokens@6`.

Examples can be found in `steps-run-terraform.yaml`

**Review by jrabbott — CHANGES_REQUESTED** [review]

File: web/pipelines/build.yaml (hunk 1 above)
Comment (Faizi52996) [reply web/pipelines/build.yaml]: changed to testing configurations.

File: web/pipelines/steps-web-run-e2e-tests.yaml (hunk 3 above)
Comment (Faizi52996) [reply web/pipelines/steps-web-run-e2e-tests.yaml]: done

Comment (jrabbott) [reply web/pipelines/steps-web-run-e2e-tests.yaml]: Please see the examples - the inputs are different between v3 and v6

File: web/pipelines/steps-web-run-a11y-tests.yaml (hunk 2 above)
Comment (jrabbott) [reply web/pipelines/steps-web-run-a11y-tests.yaml]: Please see the examples - the inputs are different between v3 and v6

Comment (Faizi52996) [reply web/pipelines/steps-web-run-a11y-tests.yaml]: ahh yes. i have updated it now.

**Review by jrabbott — APPROVED** [review]

File: web/pipelines/steps-web-run-e2e-tests.yaml (hunk 3 above)
Comment (Faizi52996) [reply web/pipelines/steps-web-run-e2e-tests.yaml]: updated it accordingly.

**Pull request merged**